pkg/cfg/cfgmgr.go

```diff
@@ -147,16 +149,87 @@ func (c *CfgMgr) getValue(key string) (string, error) {
 	return "", fmt.Errorf("Key value for key %s not found", key)
 }
 
-func (c *CfgMgr) getRandomPassword() string {
-	digits := "0123456789"
-	lowerChars := "abcdefghijklmnopqrstuvwxyz"
-	upperChars := "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-	allChars := lowerChars + upperChars + digits
+func (c * CfgMgr) getRandomPassword() string {
+    rand.Seed(time.Now().UnixNano())
+    digits := "0123456789"
+    lowerChars := "abcdefghijklmnopqrstuvwxyz"
+    upperChars := "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    allChars := lowerChars + upperChars + digits
+    const max = 16
+    out := make([]byte, max)
+    for i := 0; i < max; i++ {
+        out[i] = allChars[rand.Intn(len(allChars))]
+    }
+    return string(out)
+}
 
-	const max = 16
-	out := make([]byte, max)
-	for i := 0; i < max; i++ {
-		out[i] = allChars[rand.Intn(len(allChars))]
-	}
-	return string(out)
+func (c *CfgMgr) CreateDB(serviceName string) error {
+    zap.L().Debug("Creating DB for serviceName ", zap.String("serviceName", serviceName))
+    dbserver, err := c.getValue(fmt.Sprintf("%s/keystone/dbserver_key", c.CustomerKeyPrefix))
+    if err != nil {
+        return err
+    }
+    host, err := c.getValue(fmt.Sprintf("%s/host", dbserver))
+    if err != nil {
+        return err
+    }
+    port, err := c.getValue(fmt.Sprintf("%s/port", dbserver))
+    if err != nil {
+        return err
+    }
+    adminUser, err := c.getValue(fmt.Sprintf("%s/admin_user", dbserver))
+    if err != nil {
+        return err
+    }
+    adminPass, err := c.getValue(fmt.Sprintf("%s/admin_pass", dbserver))
+    if err != nil {
+        return err
+    }
+    dbObject, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%s)/", adminUser, adminPass, host, port))
+    if err != nil {
+        zap.L().Error("Can't connect to MySQL")
+        return err
+    }
+    _, err = dbObject.Exec(fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", serviceName))
+    if err != nil {
+        zap.L().Error("Error while creating database")
+        return err
+    }
+    zap.L().Info("Created DB successfully")
+    rows, err := dbObject.Query("SELECT @@hostname")
+    if err != nil {
+        zap.L().Error("Error while getting hostname")
+        return err
+    }
+    defer rows.Close()
+    var hostname string
+    for rows.Next() {
+        _ = rows.Scan(&hostname)
+    }
+    dbPassword := c.getRandomPassword()
+    hosts := []string{"localhost", "%", hostname}
+    for _, host := range hosts {
+        query := fmt.Sprintf("GRANT ALL PRIVILEGES ON %s.* TO '%s'@'%s' IDENTIFIED BY '%s'",
+                             serviceName, serviceName, host, dbPassword)
```

Comment:
take userName as an argument (you can keep it optional, and default to serviceName)